Patch-Source: https://gitlab.postmarketos.org/postmarketOS/tinydm/-/commit/070d51c0142643055bdee520ad954706854e2562
---
From 070d51c0142643055bdee520ad954706854e2562 Mon Sep 17 00:00:00 2001
From: Stefan Hansson <newbyte@postmarketos.org>
Date: Sat, 22 Nov 2025 11:34:30 +0100
Subject: [PATCH] systemd: Add service file

Taken from the systemd-services package in pmaports.

Closes https://gitlab.postmarketos.org/postmarketOS/tinydm/-/issues/11

Signed-off-by: Stefan Hansson <newbyte@postmarketos.org>
Part-of: <https://gitlab.postmarketos.org/postmarketOS/tinydm/-/merge_requests/25>
---
 Makefile       |  6 ++++++
 tinydm.service | 48 ++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 54 insertions(+)
 create mode 100644 tinydm.service

diff --git a/Makefile b/Makefile
index 0015670..45c1eba 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@
 
 DESTDIR :=
 OPENRC	:= 1
+SYSTEMD := 1
 
 install:
 	install -Dm755 tinydm-run-session.sh \
@@ -18,3 +19,8 @@ install:
 		install -Dm644 tinydm.confd \
 			"$(DESTDIR)/etc/conf.d/tinydm"; \
 	fi
+
+	if [ "$(SYSTEMD)" = "1" ]; then \
+		install -Dm644 tinydm.service \
+			-t "$(DESTDIR)/usr/lib/systemd/system"; \
+	fi
diff --git a/tinydm.service b/tinydm.service
new file mode 100644
index 0000000..7c83903
--- /dev/null
+++ b/tinydm.service
@@ -0,0 +1,48 @@
+[Unit]
+Description=Tiny wayland / x11 session starter for single user machines
+
+# replaces the getty
+Conflicts=getty@tty1.service
+After=getty@tty1.service dbus.socket
+
+# Needs all the dependencies of the services it's replacing
+# (currently getty@tty1.service):
+After=rc-local.service plymouth-quit-wait.service systemd-user-sessions.service
+
+# This scope is created by pam_systemd when logging in as the user.
+# This directive is a workaround to a systemd bug, where the setup of the
+# user session by PAM has some race condition, possibly leading to a failure.
+# See README for more details.
+After=session-c1.scope
+
+OnFailure=getty@tty1.service
+
+# Prevent starting on systems without virtual consoles
+ConditionPathExists=/dev/tty0
+
+[Service]
+ExecStart=tinydm-run-session
+User=10000
+Group=10000
+PAMName=login
+WorkingDirectory=~
+Restart=always
+RestartSec=1s
+
+# A virtual terminal is needed.
+TTYPath=/dev/tty1
+TTYReset=yes
+TTYVHangup=yes
+TTYVTDisallocate=yes
+
+# Fail to start if not controlling the tty.
+StandardInput=tty-fail
+StandardOutput=journal
+StandardError=journal
+
+# Log this user with utmp, letting it show up with commands 'w' and 'who'.
+UtmpIdentifier=tty1
+UtmpMode=user
+
+[Install]
+Alias=display-manager.service
-- 
GitLab

